director/advertise.go

```diff
@@ -42,27 +41,25 @@ func parseServerAd(server utils.Server, serverType server_structs.ServerType) se
 	// but endpoints do not have a scheme. As such, we need to add one for the.
 	// correct parsing. Luckily, we don't use this anywhere else (it's just to
 	// make the url.Parse function behave as expected)
-	if !strings.HasPrefix(server.Endpoint, "http") { // just in case there's already an http(s) tacked in front
-		server.Endpoint = "http://" + server.Endpoint
+	serverUrl, err := url.Parse(server.Endpoint)
+	if err != nil {
+		log.Warningf("Namespace JSON returned server %s with invalid unauthenticated URL %s",
+			server.Resource, server.Endpoint)
 	}
+	serverUrl.Scheme = "http"
```

Comment:
Looking at the comment around 40-43, the logic here is to make sure `url.Parse` work properly. However, my code snippet suggests that `url.Parse` works without the source string having a scheme: https://goplay.tools/snippet/0JAxY3I5hnt

Could you confirm this and remove the comment if so? Also I think we should set `https` instead of `http` (not 100% sure, so we probably want to check with Justin to confirm if the topology servers support https)

Reply:
Unfortunately, in general the non-auth URL for origins/caches from topology are only set up to work with `http`. For example, the origin listed in topology with this endpoint `unl-origin.nationalresearchplatform.org:1094` gets a `403` when you curl with `http`, but gets a `refused connection` when you try to curl with `https`. 

This is all due to something I don't fully understand from the way things work in the legacy OSDF deployments. In general, though, all Pelican servers should be on HTTPS. Do we have a way to determine whether a particular origin/cache is Pelican or legacy? That might be really useful in the serverAd if not

Reply:
Since setting `http` for the `serverUrl` is what we had before, let's simply remove the old comment in line 40-43 and instead note what Justin reported here and we should be good for now